server/api/profiles.js

```diff
@@ -38,16 +38,24 @@ router.post(
   }
 );
 
-router.get("/", (req, res) => {
-  getAllProfiles()
-    .then((data) => {
-      res.send(data);
-    })
-    .catch((err) => {
-      console.error(err);
-      res.send(500);
-    });
-});
+router.get(
+  "/",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    if (req.user.role === "recruiter") {
+      getAllProfiles()
+        .then(data => {
+          res.send(data);
+        })
+        .catch(err => {
+          console.error(err);
+          res.send(500);
+        });
+    } else {
+      res.sendStatus(403);
+    }
```

Comment:
You might want to include more information like before e.g. `res.status(403).send("you failed authorisation")` however this is not essential.

Reply:
If we want to add a more useful message, maybe "user in token in authorisation header must have the recruiter role" (403 already tells us we failed authorisation)